doc: Explain payload formatter types (#193)
TheThingsIndustries/lorawan-stack-docs: doc/content/integrations/payload-formatters/_index.md

Payload formatters allow you to process data going to and from end devices. This is useful for converting binary payloads to human readable fields, or for doing any other kind of data conversion on uplinks and downlinks.

This section explains how to set up payload formatters for a specific end device or an entire application.

<!--more-->

## Types of Payload Formatters

{{% tts %}} supports three types of Payload Formatters: Javascript, CayenneLPP, and Device Repo payload formatters.

### Javascript

> {{% tts %}} allows you to write your own custom Payload Formatters in Javascript. To find out how to write your own custom payload formatters, see the [Javascript Payload Formatters section]({{< relref "javascript" >}}).

Reviewer: ```suggestion
To find out how to write a custom Javascript payload formatter, see the [Javascript Payload Formatters]({{< relref "javascript" >}}) section.
```

### CayeneLPP

> {{% tts %}} can automatically decode CayenneLPP formatted payloads, no custom code neccessary. To find out how to use CayenneLPP Payload Formatters, see the [CayenneLPP section]({{< relref "cayenne" >}}).

Reviewer: ```suggestion
{{% tts %}} can automatically decode CayenneLPP formatted payloads, no custom code neccessary. To find out how to use CayenneLPP payload formatters, see the [CayenneLPP]({{< relref "cayenne" >}}) section.
```

### Device Repository

> Device manufacturers may submit payload formatters designed to work with their devices. These are publicly available in {{% tts %}} [Device Repository](https://github.com/TheThingsNetwork/lorawan-devices/tree/master).

Reviewer: Where exactly?

Reviewer 2: Would be nice to link to a guide on how to contribute payload formatters.

Author: This is coming with https://github.com/TheThingsNetwork/lorawan-stack/issues/3522
